### ISSUE TITLE
Fast forwarding a lot causes touch input to freeze melonDS temporarily

### ISSUE THREAD

**Issue by benduckwdesign** [melonDS-emu/melonDS#1563]
OS: Windows
melonDS 0.9.5 (latest release)

After fast forwarding a lot, it is impossible to use touch input by clicking in the window. Clicking in the window will render melonDS unresponsive for a short period of time (game audio still plays, but frame is frozen). After waiting, melonDS will resume but trying to touch again will freeze again. The only solution I have found is to restart the melonDS executable entirely. There are edge cases where after running for a long period of time (8+ hours,) instead of becoming unresponsive, melonDS will crash entirely, however I have not been able to reproduce the crash since the first time I encountered it.

Possible reproduction can be found by playing Pokemon Platinum. Fast forward, while holding down the fast forward Joystick hotkey (set to right trigger on DS5,) for ~8000 in-game steps. Release the trigger, then try to click anywhere on the touch screen (change Poketch screen, for instance.) This issue is probably not isolated to Pokemon, but it is possible due to its heavy reliance on actual time.

DLDI is disabled.
Crash encountered using JIT recompiler with Software 3D renderer (separate thread.)
Freezes encountered with JIT recompiler off with Software 3D renderer (separate thread.)
Unknown if JIT recompiler actually affects outcome.
Using external firmware.
Limit framerate is on.
Audio set to Cubic 16-bit.
Multiplayer audio output set to 1 instance only (multiplayer not used.)
Screen rotation and gap set to 0.
Layout is hybrid with emphasize top.
Boot game directly is enabled, but game was started indirectly by selecting File > Boot firmware after loading cartridge data.

Let me know if I can be of further assistance.

**Comment by nadiaholmquist** [comment]
This sounds really weird.

Tried to reproduce it and I can't get it to happen on macOS, so maybe this is a Windows-specific issue.

**Comment by benduckwdesign** [comment]
Was able to reproduce first try on Linux. However, on Linux, after a while, the touch does stabilize again so restarting is not necessary.

![dddcbe37-7c74-4dca-9c0e-c3130d347591](https://user-images.githubusercontent.com/5730541/202528354-948df386-afff-4ad6-8737-84cb6a5e76ce.png)

[log.zip](https://github.com/melonDS-emu/melonDS/files/10034527/log.zip)

Keyboard / joystick input also seems to be exaggerated after fast forwarding which makes me think melonDS is capturing? too many inputs while fast forwarding and then having a hard time keeping up with them afterwards, but I am probably wrong.

**Comment by patataofcourse** [comment]
if it happens in Windows and Linux, but not Mac, this may be a Qt5 issue?

**Comment by benduckwdesign** [comment]
Can confirm that the fast forward bug does not appear to happen on macOS.
However, all around performance was pretty bad on non-M1 macOS (Pro 2016? 2015?). enlarging the window or full-screening even with Software render (1x) absolutely tanks performance and makes the entire app unresponsive, though somehow 1x at tiny native window size is OK.
Also encountered a bug on macOS where on some launches there was no audio, probably related to #1564 .

Tried to reproduce in the short term on Windows with JIT enabled, but wasn't possible. Seems to be most reproducible with JIT off, although I'm not ruling out a sneaky JIT problem because of the original crash I experienced during long (8+ hour) play.

**Comment by LOuroboros** [comment]
I'm noticing by rushing through the beginning of a new savefile in Pokémon HeartGold a similar issue, possibly the same.

Basically, I'd use the fast forward, and sooner rather than later, MelonDS' window would become very unresponsive. It's happening right now as a matter of fact.

The directional keys get stuck, I can't put MelonDS' window back in foreground, it's just not responding.

After a couple of tests to find out what could be causing it, I noticed that this only seems to happen when the window is maximized.

This helped me notice something else soon enough; the emulator seems to perform worse the more you increase its window size...?

https://user-images.githubusercontent.com/4485172/203889939-58894677-0d9f-4075-8c8c-68c052692a20.mp4

https://user-images.githubusercontent.com/4485172/203889948-53f040c4-3bb4-4799-a706-e7fc04fefee3.mp4

**Comment by benduckwdesign** [comment]
Yeah, the window size definitely seems to have a bigger impact than it should have, especially when I tested on macOS. It’s not noticeable that much on my main PC, but it’s still there. Would be interesting if window size was having a direct impact on this issue. I think my mac is close to 3000x2000? And my pc is standard 2160p and I do play maximized on pc so this could have something to do with its performance. I could only test at native window size (1x) on macOS because of slowdown so it’s possible that the issue doesn’t occur unless the window is scaled, I can do a retest soon.

**Comment by benduckwdesign** [comment]
Confirmed that a larger window does seem to make the issue worse on Linux as well (used same settings as first test with 1x software render), especially with responding to inputs, although I didn't test on Linux long enough to get "not responding" again. Looks like the issue is definitely independent of platform and linked somehow to window size.

Still using 0.9.5 but my system specs have changed slightly so I will list what has changed here. I doubt GPU drivers matter but listed them just in case.

```
OS: Ubuntu 22.10 x86_64
Kernel: 6.1.12-3-liquorix-amd64
DE: GNOME 43.1
...
NVIDIA-SMI 525.89.02    Driver Version: 525.89.02    CUDA Version: 12.0
```

**Comment by benduckwdesign** [comment]
> does enabling or disabling OpenGL display make any difference?

Using OpenGL display checked with Software 3D renderer does seem to rectify the issue at least on Linux. With OpenGL display disabled the fastforward struggles to get over 120 fps, can easily obtain 300fps with the option enabled.

**Comment by LOuroboros** [comment]
> does enabling or disabling OpenGL display make any difference?

In my case, enabling OGL Display caps the speed to 100% even if I leave the Fast Forward functionality enabled. I noticed this happens a result of having VSync force-enabled in the NVIDIA Control Panel to prevent stuttering..

If I make an exception in the NCP for MelonDS, and inside of it I use the Software Renderer + OGL Display, it seems that the issue effectively goes away 😄
